src/stan/model/indexing/rvalue.hpp

```diff
@@ -18,28 +18,44 @@ namespace model {
 // all indexing from 1
 
 /**
- * Return the result of indexing a specified value with
- * a nil index list, which just returns the value.
+ * Return the result of indexing a specified scalar type with
+ * a nil index list, which just returns the scalar.
  *
  * Types:  T[] : T
  *
  * @tparam T Scalar type.
  * @param[in] c Value to index.
  * @return Input value.
  */
-template <typename T>
-inline T rvalue(const T& c, const nil_index_list& /*idx*/,
-                const char* /*name*/ = "", int /*depth*/ = 0) {
+template <typename T, typename = require_stan_scalar_t<T>>
+inline T rvalue(T&& c, const nil_index_list& /*idx*/, const char* /*name*/ = "",
```

Comment:
Why is here forwarding reference? I think you used pass by value for scalars in other places.

Reply:
Yep good catch!